lib/utils/location.js

```diff
@@ -75,17 +75,17 @@ const appPageTypes: { [AppType]: {| default?: PageType, pageTypes: PageType[] |}
 		pageTypes: ['wiki', 'search', 'stylesheet', 'modqueue', 'subredditAbout', 'comments', 'commentsLinklist', 'profile', 'liveThread', 'inbox', 'submit', 'account', 'prefs'],
 	},
 	d2x: {
-		pageTypes: ['profile2x', 'profileCommentsPage'],
+		pageTypes: ['profile2x', 'profileCommentsPage', 'submit'],
 	},
 	options: {
 		pageTypes: [],
 	},
 };
 
-export const pageType = _.once((): ?PageType => {
```

Comment:
To avoid making this expensive (as `isPageType` is invoked under many circumstances), could `pageType` be made a variable which is updated on `reddit.urlChanged`? For example: 

```
expert let pageType;
(function updatePageType() { 
  pageType =  ...; 
  waitForEvent('reddit.urlChanged').then(updatePageType);
})()
```

Reply:
Yeah, good catch. Thanks!

Reply:
Okay, so I have been trying to get this to work. The code as above works fine in the browser. But, the tests fail. What I found out:
1) jQuery ($) is being imported in `utils/dom.js`, and the error `window is not defined` pops up when `waitForEvent` is imported.

2) If I emulate the functionality in `waitForEvent` in `utils/location.js`, I get this error,
![Screenshot from 2019-03-19 02-05-40](https://user-images.githubusercontent.com/20063271/54562171-72e17b00-49ec-11e9-9620-6e57e1da0f4d.png)

I am pushing the state of the code as at point 2. I'd appreciate help in debugging this.

Reply:
This will actually break the tests for `utils/location.js` because getting the page type will dive into the DOM. And since the function is an IIFE, it will be run when importing functions from the file.

Reply:
In that case, maybe this pattern could be used: https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/e52dd41196b0cbd597d137ed1703230341b886af/lib/utils/dom.js#L142-L143